src/utils/clipboard.js

```diff
@@ -0,0 +1,62 @@
+function getClipboard() {
+    if (Clipboard.instance) {
+        return Clipboard.instance;
+    }
+
+    new Clipboard();
+
+    return Clipboard.instance;
+}
+
+class Clipboard {
+    constructor() {
+        if (Clipboard.instance) {
+            return;
+        }
+
+        this.clipboardRead = false;
+        this.clipboardWrite = false;
+
+        this.init();
+
+        Clipboard.instance = this;
+    }
+
+    // methods
+    init() {
+        navigator.permissions.query({ name: 'clipboard-write' })
+            .then(result => {
+                if (result.state == 'granted' || result.state == 'prompt') {
+                    this.clipboardWrite = true;
+                }
+            });
+
+        navigator.permissions.query({ name: 'clipboard-read' })
+            .then(result => {
+                if (result.state == 'granted' || result.state == 'prompt') {
+                    this.clipboardRead = true;
+                }
+            });
+    }
+
+    updateClipboard(newClip) {
+        if (!this.clipboardWrite) {
+            return Promise.reject('Error: No permission to write clipboard');
+        }
+
+        // return promise
+        return navigator.clipboard.writeText(newClip);
+    }
+
+    readClipboard() {
+        if (!this.clipboardRead) {
+            return Promise.reject('Error: No permission to read clipboard');
+        }
+
+        // return promise
+        return navigator.clipboard.readText();
+    }
+}
```

Comment:
if we do const clipBoard = new Clipboard(), then return clipBoard in the getClipboard function, we dont need to check if there is an instance already. That way we can make sure its a singleton returned whenever its called.

Reply:
yeah was doing that before, but changed due to https://eslint.org/docs/rules/no-constructor-return

Reply:
According to the no-constructor-return rule, we should not be returning out of the constructor, and that is exactly what you are doing on line 14.

The comment I wrote earlier still stands. Let me know if it doesnt make sense.

Reply:
Actually its `Disallow returning value` but it's ok to have `return`.
Lets have a call, I think I am missing the point you are trying to make.